xai-model-provider/main.go

```diff
@@ -2,24 +2,65 @@ package main
 
 import (
 	"fmt"
+	"log"
+	"net/http"
 	"os"
+	"strings"
+	"time"
 
-	"github.com/obot-platform/tools/xai-model-provider/server"
+	"github.com/obot-platform/tools/openai-model-provider/proxy"
 )
 
+const loggerPath = "/tools/xai-model-provider/validate"
+
+func ValidateXAIAPIKey(cfg *proxy.Config) error {
+	if cfg.APIKey == "" {
+		const msg = "Invalid xAI Credentials"
+		log.Printf("time=%q level=error msg=%q logger=%s", time.Now().Format(time.RFC3339), msg, loggerPath)
```

Comment:
Could as well export the log functions in the proxy package and re-use them here (if you don't want to switch to e.g. slog with a less custom format)